internal/lsp/hover.go

```diff
@@ -0,0 +1,174 @@
+package lsp
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/olekukonko/tablewriter"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/types"
+)
+
+type BuiltinPosition struct {
+	Builtin *ast.Builtin
+	Line    uint
+	Start   uint
+	End     uint
+}
+
+var builtins = builtinMap() //nolint:gochecknoglobals
+
+func builtinMap() map[string]*ast.Builtin {
+	m := make(map[string]*ast.Builtin)
+	for _, b := range ast.CapabilitiesForThisVersion().Builtins {
+		m[b.Name] = b
+	}
+
+	return m
+}
+
+func builtinCategory(builtin *ast.Builtin) (category string) {
+	if len(builtin.Categories) == 0 {
+		if s := strings.Split(builtin.Name, "."); len(s) > 1 {
+			category = s[0]
+		} else {
+			category = builtin.Name
+		}
+	} else {
+		category = builtin.Categories[0]
```

Comment:
What are we losing here, presumably the rest of the categories are not useful?

Reply:
This is to try and mimic the way categories are placed under headings in the OPA reference docs. The "other side" of this is where it's being generated, i.e. [here](https://github.com/open-policy-agent/opa/blob/main/internal/cmd/genbuiltinmetadata/main.go#L153).